### ISSUE TITLE
Gateways on devnet not working correctly

### ISSUE THREAD

**Issue by mohamedamer453** [threefoldtech/grid_weblets#964]
### Description

One of the gateways on devnet is not working correctly. i was deploying a subsquid instance on devnet and after it was successfully deployed i tried to access the instance but it didn't work. however i tried a couple of other instances that were deployed on different gateway nodes and these instances worked without any issues.

the problem appears to be coming from node `45` which is a gateway node and the instance that didn't work had the same gateway `devops.tfcloud.us`

![image](https://user-images.githubusercontent.com/101194226/186390917-3060eb2f-8e40-4269-9977-cedcaa1e2871.png)

![image](https://user-images.githubusercontent.com/101194226/186390833-e7b95e82-2661-403a-9184-964b752b73db.png)

and i verified that the issue with this instance was from the gateway since i was able to acces the same instance from the `yigg` ip and it worked without any issues.

![image](https://user-images.githubusercontent.com/101194226/186392189-52cf2a94-7b09-4f98-b8da-c8e4c835d461.png)

and the 2 working instances were deployed on gateway nodes `14` (gent01.dev.grid.tf) and `15` (gent02.dev.grid.tf)

![image](https://user-images.githubusercontent.com/101194226/186391759-46f6409c-34f2-4cd4-814c-7733d694f0a3.png)

![image](https://user-images.githubusercontent.com/101194226/186391825-33d1e4f1-3c11-413b-a1ee-9bb12ec96097.png)

**Comment by coesensbert** [comment]
Farm 59 on devnet is managed by a community member, I contacted him about the issue.

**Comment by Parkers145** [comment]
I will look into this today!

**Comment by Parkers145** [comment]
Thank you for reporting this, my dns record had an error in it and has been corrected the dns now properly resolves. 

It may take up to an hour for the updates to be 100% propagated but if you wouldn't mind Re deploying I'd appreciate it!

**Comment by ramezsaeed** [comment]
@mohamedamer453 
please re-verify

**Comment by mohamedamer453** [comment]
`Node 45` is currently offline.

![image](https://user-images.githubusercontent.com/101194226/187647143-20e1f0f7-ce39-4c40-8b2d-5b4e3d3e28cd.png)

will re-verify this issue once the node is back online.

**Comment by Parkers145** [comment]
Had a mishap during some work on the neighborhood by the isp yesterday, tech should be here within the next couple hours to get it back up.

**Comment by Parkers145** [comment]
All nodes back online!

**Comment by mohamedamer453** [comment]
Unfortunately the same issue is still present. i tried a couple of different deployments on `devops.tfcloud.us` (node `45`)

and both of them did not work.

- subsquid

   ![image](https://user-images.githubusercontent.com/101194226/187718360-5e3e8508-b283-4fec-91bc-aec2791df7b5.png)

- peertube

   ![image](https://user-images.githubusercontent.com/101194226/187718449-b0a026bf-abcc-4e6d-821d-a348284e791c.png)

and i also found another node that belonged to the same farm (node `49` ) `libra.tfcloud.us`

i tried a couple of other deployments on it but it also did not work.

- funkwhale

   ![image](https://user-images.githubusercontent.com/101194226/187719137-fe3437e1-e2cc-4b9b-ba1a-543f87b127b7.png)

- ownCloud

   ![image](https://user-images.githubusercontent.com/101194226/187719750-53608de1-e8a6-4c53-8dc6-3e30a1baa599.png)

**Comment by Parkers145** [comment]
![787934E2-1E71-4131-A957-E72CA01EAC1C](https://user-images.githubusercontent.com/44621168/187723783-a7cf85e5-99d3-4762-9a8c-6f020ab19641.png)

using an external dns check, the devops.tfcloud.us resolves to node 45s public address at 162.205.240.251. 

I’m not sure what more I can do to fix it.

**Comment by Parkers145** [comment]
These are my configurations perhaps im missing something?

![image](https://user-images.githubusercontent.com/44621168/187725358-9fa9cac6-af0a-4cb7-bdba-1175da643373.png)
![image](https://user-images.githubusercontent.com/44621168/187725538-c1b5607e-2ea3-401b-910b-d52d36b7b64b.png)

**Comment by mohamedamer453** [comment]
it's working now but for both domains `devops.tfcloud.us` & `libra.tfcloud.us` the connection is not secure.


![image](https://user-images.githubusercontent.com/101194226/187859458-2e7ee745-fb1a-411b-9bfd-bca51a239b83.png)

![image](https://user-images.githubusercontent.com/101194226/187859987-c6ff210a-65b9-4987-ab3d-5fa18ec10e0e.png)


![image](https://user-images.githubusercontent.com/101194226/187859638-0cae7ccc-15a4-488d-b899-d5a98e54aba3.png)

![image](https://user-images.githubusercontent.com/101194226/187860031-3b9f4d89-d231-4a2d-906b-c60df5e40a76.png)


and indeed the documentation is not enough or clear on setting up public configs & dns records for gateway nodes so i created an issue in the Threefold Manual repo to add more docs on this part.

**Comment by Parkers145** [comment]
Is the secure connection issue something I can fix?

**Comment by mohamedamer453** [comment]
I'm not really sure where the secure connection issue is originating from. but it think @coesensbert  might have an idea on how to configure/fix this.

**Comment by Parkers145** [comment]
Copy that, just let me know if I need to do anything!

**Comment by coesensbert** [comment]
> Is the secure connection issue something I can fix?

Yes, you have to have 3 kinds of DNS records for the certificate requests to Let's Encrypt to work. You need:

- A record
- CNAME (to the A record)
- NS record

So for our gent01.grid.tf domain we have:

```
A:  gent01.grid.tf -> 185.69.166.140
CNAME:  *.gent01.grid.tf -> gent01.grid.tf
NS: _acme-challenge.gent01.grid.tf -> gent01.grid.tf
```

So for yours that would be

```
A:  devops.tfcloud.us -> (PUBLIC IP OF PUBLIC GATEWAY)
CNAME:  *.devops.tfcloud.us -> devops.tfcloud.us
NS: _acme-challenge.devops.tfcloud.us -> devops.tfcloud.us
```

I also notice that that are no A records set for Devops.TFcloud.us and Libra.Tfcloud.us currently. So that would have to be done first, then the other two records.

```
➜  ~ dig Devops.TFcloud.us        

; <<>> DiG 9.18.1-1ubuntu1.1-Ubuntu <<>> Devops.TFcloud.us
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 48726
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;Devops.TFcloud.us.             IN      A

;; AUTHORITY SECTION:
TFcloud.us.             378     IN      SOA     ns43.domaincontrol.com. dns.jomax.net. 2022083114 28800 7200 604800 600

;; Query time: 16 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Sep 01 15:47:21 CEST 2022
;; MSG SIZE  rcvd: 117

➜  ~ dig Libra.Tfcloud.us         

; <<>> DiG 9.18.1-1ubuntu1.1-Ubuntu <<>> Libra.Tfcloud.us
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24974
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;Libra.Tfcloud.us.              IN      A

;; AUTHORITY SECTION:
Tfcloud.us.             546     IN      SOA     ns43.domaincontrol.com. dns.jomax.net. 2022083114 28800 7200 604800 600

;; Query time: 12 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Sep 01 15:47:25 CEST 2022
;; MSG SIZE  rcvd: 116
```

**Comment by Parkers145** [comment]
Bert your a god send, I’ll get this fixed asap

**Comment by Parkers145** [comment]
A records 
![image](https://user-images.githubusercontent.com/44621168/187962696-0e6a84ba-f022-454e-bcd9-e0b03d5bc1a3.png)

NS Records 
![image](https://user-images.githubusercontent.com/44621168/187962765-bb8bee21-c4df-41c0-b1d1-e13c83fa2fc0.png)

Cnames- 1 
![image](https://user-images.githubusercontent.com/44621168/187962849-1492d5a0-7ff2-407b-8a47-5320cc948885.png)

Cnames-2 
![image](https://user-images.githubusercontent.com/44621168/187962970-b2c3aae0-9592-4461-9841-dfb69259a8f6.png)

Hopefully we should be working now!

**Comment by Parkers145** [comment]
~$ dig libra.tfcloud.us

; <<>> DiG 9.11.3-1ubuntu1.13-Ubuntu <<>> libra.tfcloud.us
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60178
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 2, ADDITIONAL: 5

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
; COOKIE: 5591d046108d316bb07975eb6310dccf2797a026238edaad (good)
;; QUESTION SECTION:
;libra.tfcloud.us.              IN      A

;; ANSWER SECTION:
libra.tfcloud.us.       423     IN      A       162.205.240.250

;; AUTHORITY SECTION:
tfcloud.us.             3423    IN      NS      ns43.domaincontrol.com.
tfcloud.us.             3423    IN      NS      ns44.domaincontrol.com.

;; ADDITIONAL SECTION:
ns44.domaincontrol.com. 23819   IN      AAAA    2603:5:2252::16
ns43.domaincontrol.com. 23819   IN      AAAA    2603:5:2152::16
ns44.domaincontrol.com. 23819   IN      A       173.201.69.22
ns43.domaincontrol.com. 23819   IN      A       97.74.101.22

;; Query time: 53 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Thu Sep 01 11:24:38 CDT 2022
;; MSG SIZE  rcvd: 232

~$ dig devops.tfcloud.us

; <<>> DiG 9.11.3-1ubuntu1.13-Ubuntu <<>> devops.tfcloud.us
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 10152
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 2, ADDITIONAL: 5

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
; COOKIE: 9e4841abedfb69d75850faa96310dcfeb8725aef1e72c50b (good)
;; QUESTION SECTION:
;devops.tfcloud.us.             IN      A

;; ANSWER SECTION:
devops.tfcloud.us.      600     IN      A       162.205.240.251

;; AUTHORITY SECTION:
tfcloud.us.             3376    IN      NS      ns43.domaincontrol.com.
tfcloud.us.             3376    IN      NS      ns44.domaincontrol.com.

;; ADDITIONAL SECTION:
ns44.domaincontrol.com. 23772   IN      AAAA    2603:5:2252::16
ns43.domaincontrol.com. 23772   IN      AAAA    2603:5:2152::16
ns44.domaincontrol.com. 23772   IN      A       173.201.69.22
ns43.domaincontrol.com. 23772   IN      A       97.74.101.22

;; Query time: 94 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Thu Sep 01 11:25:25 CDT 2022
;; MSG SIZE  rcvd: 233

**Comment by Parkers145** [comment]
If someone could verify that these are now functioning properly, and honestly maybe run through some checks on my other nodes, I’m fully prepared to complete an entire in depth guide on configuring nodes as gateways… I just need validation that I’m putting out good info since my knowledge of on the topic is truly threefold specific I don’t want to get bad information out. 

But I can cover the topic pretty in depth at this point if we can validate my setup. I think I’m probably the first at home farm of this type and I’d love to really get it out there what’s possible. My farm is not only completely free of outside influence. I’m 100% sure that I can demonstrate that it can function entirely self sustained. 

I’m also working on a script using the daddy cli tool for godaddy where I host my domains that I’m hoping can automate the process of dns management similar to Scott’s ip script. 

Theoretically I’m sure this could be worked backwards to figure out a way to have a central dns server that accepts input directly from zos. 

either way, im a little brain split right now cause it’s my Friday at work, but if I can get confirmation that I am infact configured correctly and fully functional I’ll spend the weekend making both written and video tutorials.

**Comment by mohamedamer453** [comment]
So to make sure that the nodes are working i tried almost every solution that needs a gateway `peertube, funkwhale, ownCloud, subsquid, casperlabs, etc ...` and the connection was secure for the 2 domains `libra.tfcloud.us` & `devops.tfcloud.us` 


however i encountered a couple of weird cases, the first one is related to `peertube` and i encountered this using the 2 domains. after successfully deploying 2 instances one on each gateway node i got the following error when i tried to access these instances. 

![image](https://user-images.githubusercontent.com/101194226/188307153-fdec55c8-4a5f-49ca-8710-98855e9602d0.png)

and when i tried to login using admin email/password or using `ThreeFold Connect` i got the same error.

![image](https://user-images.githubusercontent.com/101194226/188307235-7c838073-7105-45a4-b2bf-91940ff66d5b.png)


and to verify that this issue was only related to these domains i then deployed another peertube instance but this time i deployed it on `gent01.dev.grid.tf` and it worked and the login was working using both methods.


The second case i encountered only while using `libra.tfcloud.us`. while using this domain the deployed instances gave `Bad Gateway` for a while and then these instance worked after a while (for some it was only a couple of minutes and for others it was 15+ mins) and now i have a `subsquid` instance stuck on `Bad Gateway` for almost an hour now.


![image](https://user-images.githubusercontent.com/101194226/188307717-3d903bb2-61d0-4618-a9fc-820a169b7b63.png)

**Comment by Parkers145** [comment]
I’m not familiar with either of those apps, 

i have one gremlin in my network still that seems to cause paroxysmal dhcp failures. I think I have tracked this down to a compatability with an sfp+ module and 5gbase-t,  unfortunately I don’t have a solution yet.

**Comment by coesensbert** [comment]
Deployments work for me too now, great job @Parkers145 !

**Comment by mohamedamer453** [comment]
Verified.

Both domains are now working with different deployments without any issues.

![image](https://user-images.githubusercontent.com/101194226/188927483-f1148f59-89b1-45f2-8bff-56861952eecd.png)

![image](https://user-images.githubusercontent.com/101194226/188927599-9d14143f-1785-4581-b875-aa4df999cbaf.png)

and i created another issue for the peertube login error https://github.com/threefoldtech/grid_weblets/issues/1000
